File: 2026/Company/Company.Web/Modules/ServerTypes/Sales/CurrencyRateColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CurrencyRateRow } from "./CurrencyRateRow";

export interface CurrencyRateColumns {
    CurrencyRateId: Column<CurrencyRateRow>;
    CurrencyRateDate: Column<CurrencyRateRow>;
    FromCurrencyCode: Column<CurrencyRateRow>;
    ToCurrencyCode: Column<CurrencyRateRow>;
    AverageRate: Column<CurrencyRateRow>;
    EndOfDayRate: Column<CurrencyRateRow>;
}

export class CurrencyRateColumns extends ColumnsBase<CurrencyRateRow> {
    static readonly columnsKey = 'Sales.CurrencyRate';
    static readonly Fields = fieldsProxy<CurrencyRateColumns>();
}

File: 2026/Company/Company.Web/Modules/Sales/CurrencyRate/CurrencyRateColumns.cs
namespace Company.Sales.Columns;

[ColumnsScript("Sales.CurrencyRate")]
[BasedOnRow(typeof(CurrencyRateRow), CheckNames = true)]
public class CurrencyRateColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int CurrencyRateId { get; set; }
    public DateTime CurrencyRateDate { get; set; }
    [EditLink]
    public string FromCurrencyCode { get; set; }
    public string ToCurrencyCode { get; set; }
    public decimal AverageRate { get; set; }
    public decimal EndOfDayRate { get; set; }
}